spring-data-geode/src/test/java/org/springframework/data/gemfire/function/execution/onservers/FunctionsReturnResultsFromAllServersIntegrationTests.java

```diff
@@ -0,0 +1,172 @@
+public class FunctionsReturnResultsFromAllServersIntegrationTests extends ClientServerIntegrationTestsSupport {
+
+	private static final int PORT_1 = 40407;
+	private static final int PORT_2 = 40403;
+
+	private static ProcessWrapper gemfireServer1;
+	private static ProcessWrapper gemfireServer2;
+
+	@Autowired
+	private AllServersAdminFunctions allServersAdminFunctions;
+
+	@Autowired
+	private SingleServerAdminFunctions singleServerAdminFunctions;
+
+	@BeforeClass
+	public static void startGemFireServer() throws Exception {
+
+		gemfireServer1 = run(MetricsFunctionServerProcess.class,
+				String.format("-D%s=%d", GEMFIRE_CACHE_SERVER_PORT_PROPERTY, PORT_1));
+
+		waitForServerToStart(DEFAULT_HOSTNAME, PORT_1);
+
+		gemfireServer2 = run(MetricsFunctionServerProcess.class,
+				String.format("-D%s=%d", GEMFIRE_CACHE_SERVER_PORT_PROPERTY, PORT_2));
+
+		waitForServerToStart(DEFAULT_HOSTNAME, PORT_2);
+	}
+
+	@ClientCacheApplication(servers = {@ClientCacheApplication.Server(port = PORT_1), @ClientCacheApplication.Server(port = PORT_2)})
+	@Configuration
+	@EnableGemfireFunctionExecutions(basePackageClasses = AllServersAdminFunctions.class)
+	@EnableGemfireCacheTransactions
+	static class Config { }
+
+	@AfterClass
+	public static void stopGemFireServer() {
+		stop(gemfireServer1);
+		stop(gemfireServer2);
+	}
+
+	@Test
+	public void executeFunctionOnAllServers() {
+		List<List<Metric>> metrics = allServersAdminFunctions.getAllMetrics();
+		assertThat(metrics.size()).isEqualTo(2);
+	}
+
+	@Test
+	public void executeFunctionOnSingleServer() {
+		List<Metric> metrics = singleServerAdminFunctions.getAllMetrics();
+		assertThat(metrics.size()).isEqualTo(672);
+	}
+
+	static class MetricsFunctionServerProcess {
+
+		private static final int DEFAULT_CACHE_SERVER_PORT = 40404;
+
+		private static final String CACHE_SERVER_PORT_PROPERTY = "spring.data.gemfire.cache.server.port";
+		private static final String GEMFIRE_LOG_LEVEL = "error";
+		private static final String GEMFIRE_NAME = "MetricsServer" + getCacheServerPort();
+
+		public static void main(String[] args) throws Exception {
+			registerFunctions(startCacheServer(newGemFireCache()));
+		}
+
+		private static Cache newGemFireCache() {
+
+			return new CacheFactory()
+					.set("name", GEMFIRE_NAME)
+					.set("log-level", GEMFIRE_LOG_LEVEL)
+					.create();
+		}
+
+		private static Cache startCacheServer(Cache gemfireCache) throws IOException {
+			CacheServer cacheServer = gemfireCache.addCacheServer();
+			cacheServer.setPort(getCacheServerPort());
+			cacheServer.start();
+			return gemfireCache;
+		}
+
+		private static int getCacheServerPort() {
+			return Integer.getInteger(CACHE_SERVER_PORT_PROPERTY, DEFAULT_CACHE_SERVER_PORT);
+		}
+
+		private static Cache registerFunctions(Cache gemfireCache) {
+
+			FunctionService.registerFunction(new GetAllMetricsFunction());
+
+			return gemfireCache;
+		}
+	}
```

Comment:
Let's stay clear of using the Geode API to configure and bootstrap a Geode Server.  We should be using SDG's API, and preferably the annotation configuration model to do the same.

For [instance](https://github.com/spring-projects/spring-data-geode/blob/master/spring-data-geode/src/test/java/org/springframework/data/gemfire/config/annotation/EnableSslConfigurationIntegrationTests.java#L121-L171).

Reply:
Using a Springier way to configure the server